starter-cli/src/main/java/io/micronaut/starter/cli/util/openrewrite/maven/MavenOpenRewriteRecipesRunner.java

```
@@ -0,0 +1,79 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.cli.util.openrewrite.maven;
+
+import io.micronaut.starter.cli.util.openrewrite.OpenRewriteConfiguration;
+import io.micronaut.starter.cli.util.openrewrite.OpenRewriteRecipesRunner;
+import io.micronaut.starter.application.OperatingSystem;
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+import org.apache.maven.shared.invoker.*;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+@Named("maven")
+@Singleton
+public class MavenOpenRewriteRecipesRunner implements OpenRewriteRecipesRunner {
+
+    private static final String SYS_PROPERTY_REWRITE_RECIPE_ARTIFACT_COORDINATES = "rewrite.recipeArtifactCoordinates";
+    private static final String DEFAULT_MICRONAUT_REWRITE_RECIPE_ARTIFACT_COORDINATES = "io.micronaut.openrewrite:micronaut-openrewrite:LATEST";
+    private static final String DEFAULT_REWRITE_MAVEN_PLUGIN_VERSION = "6.19.0";
+
+    @Override
+    public void run(List<String> recipes,
+                    File folder,
+                    OpenRewriteConfiguration configuration,
+                    Consumer<String> out,
+                    Consumer<String> err) {
+        InvocationRequest request = createInvocationRequest(folder, configuration, out, err);
+        Invoker invoker = new DefaultInvoker();
+        try {
+            invoker.execute(request);
+        } catch (MavenInvocationException e) {
+            throw new RuntimeException(e);
```

How is this rendered to the user?

---

MavenOpenRewriteRecipesRunner sets request.setOutputHandler(out::accept) and request.setErrorHandler(err::accept) then UpdateCommand passes this::out and this::err into runner.run(...)

---

This doesn't answer my question. My question is: if a `MavenInvocationException` occurs, what is what the user gets in the terminal? A stacktrace? Because if that is the case, then I think we should better handle exceptions and not let them propagate all the way through the users

---

i guess this will prevent stacktrace https://github.com/micronaut-projects/micronaut-starter/pull/2850/files#r2413867745